src/libraries/System.IO.MemoryMappedFiles/src/Microsoft/Win32/SafeMemoryMappedFileHandle.Unix.cs

```diff
@@ -74,7 +85,17 @@ protected override void Dispose(bool disposing)
             base.Dispose(disposing);
         }
 
-        protected override bool ReleaseHandle() => true; // Nothing to clean up.  We unlinked immediately after creating the backing store.
+        protected override bool ReleaseHandle() {
+            Debug.Assert(!IsInvalid);
```

Comment:
This assert isn't valid.  The ctor could throw before reaching the SetHandle call, in which case the handle will still have a value of 0, which is considered invalid.